Remove feature flag from task requestor details modal to implement new design by default

Hariom01010 opened the pull request:
<!-- Date Here in dd mmm yyyy-->
Date: 31 Mar 2024
<!--Developer Name Here-->
Developer Name: @Hariom01010 

---

## Issue Ticket Number
<!--Issue ticket this PR closes-->
- Closes #966
## Description
This PR aims at removing the feature flag from the Task Requestor Details modal.
Previously the new design was behind ff, this PR aims to remove the ff and make the new design the default one
<!--Description of the changes made in this PR-->

### Documentation Updated?

- [ ] Yes
- [X] No

<!--Additional notes about documentation update if applicable-->

### Under Feature Flag

- [ ] Yes
- [X] No

<!--Indicate if changes are under a feature flag-->

### Database Changes

- [ ] Yes
- [X] No

<!--Notes on any database changes-->

### Breaking Changes

- [ ] Yes
- [X] No

<!--Notes on breaking changes or related issue tickets if applicable-->

### Development Tested?

- [X] Yes
- [ ] No

<!--Confirmation of local testing during development-->

## Screenshots
<details>
<summary>Screenshot 1</summary>

<!-- Attach your screenshots here👇-->
If we look at the url now the new modal is working even without the dev=true query parameter

https://github.com/user-attachments/assets/40ba5cf1-7667-4686-b611-eb8871011046

https://github.com/user-attachments/assets/595c5c8b-2c93-4cfe-84a2-e18e5bf56ebf



![Screenshot 2025-03-28 180045](https://github.com/user-attachments/assets/1e8c16c6-d59b-4697-9194-c448dd991b87)

</details>

<!--Attach or link to relevant screenshots or visual aids-->

## Test Coverage
<details>
<summary>Screenshot 1</summary>

<!-- Attach your screenshots here👇-->
![image](https://github.com/user-attachments/assets/c8226333-aae2-4e54-a355-ac7abeda01e2)

</details>

<!--Attach Details on test coverage and outcomes-->

## Additional Notes

<!--Any additional notes, considerations or explanations for reviewers-->
The test case we have introduced to check if "N/A" is being rendered or not is because earlier when the feature flag was not there the new modal and old modal rendering was just being tested on the basis of the class names (since we were using different class names for both the model). Now since feature flag is being removed the values in old class names will be replaced with the new values, so there is not a test which confirms the new modal is being rendered . This test case helps us verify that new modal is being rendered since the "N/A" is being rendered for empty description in the new design only.

MayankBansal12 commented:
Hello @Hariom01010, please update the PR title and attach a screen recording for the change.

MayankBansal12 commented on __tests__/task-requests/task-requestDetails.test.js:
we don't need test to check the behaviour for the new modal?

Hariom01010 replied:
There is already a test case written which is check if the modal is working as expected

MayankBansal12 replied:
Share the link for the test that tests rendering `N/A` if the description is not provided, and resolve the conversation.

Hariom01010 replied:
Added the test case to handle when description is not provided and render 'N/A'. Thanks for bringing it into my notice

Hariom01010 commented:
> Hello @Hariom01010, please update the PR title and attach a screen recording for the change.

Updated the PR Title

MayankBansal12 commented:
> Updated the PR Title and added a screen recording showing the modal is working fine

Add screen recording for displaying empty description and also add test coverage

Hariom01010 commented:
> > Updated the PR Title and added a screen recording showing the modal is working fine
> 
> Add screen recording for displaying empty description and also add test coverage

Added updated test coverage and a video depicting model behavior in case of empty description

mbramani reviewed (approved):
LGTM

Hariom01010 commented on __tests__/task-requests/task-requestDetails.test.js:
```diff
@@ -159,66 +159,45 @@ describe('Task request details page', () => {
   });
 
-  it('should properly handle long descriptions in the modal', async function () {
+  it('should render N/A for description if descriptions is not present', async function () {
```
Earlier when the feature flag was not there the new modal and old modal rendering was just being tested on the basis of the class names (since we were using different class names for both the model). Now since feature flag is being removed the values in old class names will be replaced with the new values, so there is not a test which confirms the new modal is being rendered . This test case helps us verify that new modal is being rendered since the "N/A" is being rendered for empty description in the new design only.

AnujChhikara replied:
Understood , You can mention this in the Additional Notes section of the PR . Else everything looks okay

Hariom01010 replied:
Got it, will be mentioning this in the Additional Notes section. Resolving this conversationg now. Please approve the PR if everything looks good
